source/reflect.h2

```diff
@@ -1501,13 +1662,38 @@ union: (inout t : meta::type_declaration)
 
 //-----------------------------------------------------------------------
 //
-//  print - output a pretty-printed visualization of t
+//  print - output a pretty-printed visualization of t or f
 //
 print: (t: meta::type_declaration) =
 {
     std::cout << t.print() << "\n";
 }
 
+print: (f: meta::function_declaration) =
+{
+    std::cout << f.print() << "\n";
+}
+
+
+//-----------------------------------------------------------------------
+//
+//  meta - mark f as a metafunction and generate registration code for it
+//
+meta: (inout f: meta::function_declaration) =
```

Comment:
Why do we need both `in` passing style and `inout` passing style. `inout` would also cover all cases. Since both exist it can not be because of security or speed reasons. 

I would suggest to drop thee `in` passing style.

Reply:
there's currently a metafunction like `print` that uses `in` parameter style since it doesn't modify the declaration. whether or not we want to support those is up to Herb really, but since `in` is the default parameter I feel like I can't leave it out for now.

Reply:
Yes, I know. I am just asking the question: Was the `in` style because of some special thoughts or because it was just the minimal required definition for this function.